### PR TITLE
Internal order fixes

### LINKED ISSUE #317: Internal Order fixes
Coming from #225 - have broken up that issue, it's too big to review in one chunk.

**1. Enter an Internal Order from Kopu Health Center (KHC) to Opua Health Center (OHC):** 
- [x] Add Item: should the item field be activated/highlighted by default ? 
- [x] Add Item: tabulation order should be: Item > Requested Quantity > Comment > OK > Ok & Next > Cancel
- [x] Add Item: when selecting an item (eg. Paracetamol), only `target quantity` charts appears. Others are missing. If I validate and re-open, all charts are visible. 
- [x] Add Item > `Stock Evolution` chart: requested quantity should update the chart (not suggested quantity)
- [x] Add Item > `Consumption History` chart > Moving Average: chart gives the impression that the AMC is taken into account to calculate the Mov. Avg (which should not be the case)
- [x] Add Item > Buttons: when pressing `Enter` on my KB, it does not activate the highlighted button (OK, OK & Next or Cancel). I can only click on it to activate the action button. 
- [x] Print > does not seem to be working while writing up this issue

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #317 

- [ ] Update tab order to deprioritise cancel
- [ ] Creating the stock lines on item selection in order to show charts 
- [ ] Delete request line for previous item when changing item - but only when creating a request line
- [ ] Delete request line when cancelling a create dialog
- [ ] `Stock Evolution` chart is now using requested quantity not suggested quantity
- [ ] `Consumption History` chart has the 'AMC' relabelled to 'Projected' Not sure what else to do for this request 🤷 
- [ ] Dialog buttons now fire the `onClick` when pressing space or enter

The other item, of "Print > does not seem to be working" required a database migration on the demo server.

### CODE REVIEW

File: client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx [hunk 1]
```diff
@@ -93,8 +123,14 @@ export const RequestLineEdit = ({
             sx={{ paddingLeft: 4, paddingRight: 4 }}
             justifyContent="space-between"
           >
-            <ConsumptionHistory id={draft?.id || ''} />
-            <StockEvolution id={draft?.id || ''} />
+            {draft?.isCreated ? (
+              <Box display="flex" height={289} />
```
Comment (mark-prins) [inline client/packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx:127]: The height should probably just be removed. I was trying to help reduce the jankiness, and gave up. I think a little jank is ok as a trade off for having the charts show

File: client/packages/system/src/Item/api/api.ts [hunk 2]
```diff
@@ -62,7 +62,11 @@ export const getItemQueries = (sdk: Sdk, storeId: string) => ({
         key: itemParsers.toSortField(sortBy),
         isDesc: sortBy.isDesc,
         storeId,
-        filter: { ...filterBy, type: { equalTo: ItemNodeType.Stock } },
+        filter: {
+          ...filterBy,
+          type: { equalTo: ItemNodeType.Stock },
+          isVisible: true,
```
Comment (mark-prins) [inline client/packages/system/src/Item/api/api.ts:68]: I found some items had appeared in my database ( due to sync testing ) which weren't valid, and had `isVisible` false but still showed. Seems reasonable to only show visible items?

File: client/packages/system/src/Item/api/hooks/useStockItemsWithStats/useStockItemsWithStats.tsx [hunk 3]
```diff
@@ -1,12 +1,11 @@
-import { ItemNodeType, useQuery } from '@openmsupply-client/common';
+import { useQuery } from '@openmsupply-client/common';
 import { useItemApi } from '../useItemApi';
 
 export const useStockItemsWithStats = () => {
   const queryParams = {
     sortBy: { key: 'name', isDesc: false, direction: 'asc' as 'asc' | 'desc' },
     offset: 0,
     first: 1000, // TODO: remove arbitrary limit
```
Comment (mark-prins) [inline client/packages/system/src/Item/api/hooks/useStockItemsWithStats/useStockItemsWithStats.tsx:8]: the filter is already in the api call

**Review by clemens-msupply — APPROVED** [review]
Looks good and as far as I was able to test it works nicely

**Pull request merged**